WorkspaceServer/Kernel/CSharpKernelExtensions.cs

```diff
@@ -68,22 +69,50 @@ public static CSharpKernel UseNugetDirective(this CSharpKernel kernel, INativeAs
                 {
                     Handler = async context =>
                     {
-                        var refs = await restoreContext.AddPackage(package.PackageName, package.PackageVersion);
+                        var message = $"Attempting to install package {package.PackageName}";
+                        if (!string.IsNullOrWhiteSpace(package.PackageVersion))
+                        {
+                            message += $", version {package.PackageVersion}";
+                        }
+
+                        var key = message;
+                        var displayed = new DisplayedValueProduced(message, context.Command, valueId: key);
+                        context.Publish(displayed);
+
+                        var installTask = restoreContext.AddPackage(package.PackageName, package.PackageVersion);
+
+                        while ((await Task.WhenAny(Task.Delay(1000), installTask)) != installTask)
+                        {
+                            message += "...";
+                            context.Publish(new DisplayedValueUpdated(message, key));
+                        }
+
+                        message += "done!";
+                        context.Publish(new DisplayedValueUpdated(message, key));
+
+                        var result = await installTask;
                         helper?.Configure(await restoreContext.OutputPath());
-                        if (refs != null)
+
+                        if (result.Succeeded)
                         {
-                            foreach (var reference in refs)
+                            foreach (var reference in result.References)
                             {
                                 if (reference is PortableExecutableReference peRef)
                                 {
                                     helper?.Handle(peRef.FilePath);
                                 }
                             }
 
-                            kernel.AddMetadataReferences(refs);
+                            kernel.AddMetadataReferences(result.References);
+
+                            context.Publish(new DisplayedValueProduced($"Successfully added reference to package {package.PackageName}", context.Command));
+                            context.Publish(new NuGetPackageAdded(package));
+                        }
+                        else
+                        {
+                            context.Publish(new DisplayedValueProduced($"Failed to add reference to package {package.PackageName}", context.Command));
```

Comment:
When it fails, why does it fail, and is there additional information we can surface?